fenix/app/src/main/java/org/mozilla/fenix/components/metrics/MetricsStorage.kt

```diff
@@ -114,6 +131,17 @@ internal class DefaultMetricsStorage(
             Event.GrowthData.FirstUriLoadForDay -> {
                 settings.uriLoadGrowthLastSent = System.currentTimeMillis()
             }
+            Event.GrowthData.UserActivated -> {
+                settings.adjustUserActivatedSent = true
+            }
+            Event.ActivatedUserData.FirstWeekUsageActivity -> {
+                settings.adjustUseCount.increment()
+                settings.adjustUseCountLastIncrement = System.currentTimeMillis()
+            }
+            Event.ActivatedUserData.FirstWeekSearchActivity -> {
+                settings.adjustEarlySearchCount.increment()
+                settings.adjustEarlySearchCountLastIncrement = System.currentTimeMillis()
```

Comment:
Instead of updating this kind of persistent state here, perhaps we should either introduce another function like `updateDaysOfUse` or change that function to track any persistent state we are interested in. Admittedly the example I set with `FirstAppOpenForDay` and `FirstUriLoadForDay` was a little hacky, but I was unsure at the time how long this code would live.

Reply:
changed how this works with new `updatePersistentState` function for events that require counting data before sending.